### PR TITLE
Add `UHUGEINT` support for Arrow

### PR DESCRIPTION
**Pull request by nickgerrets**
This PR adds support to convert `UHUGEINT` type to the pyarrow `decimal128` type, same as with `HUGEINT`.

### CODE REVIEW

**Review by Tishj — COMMENTED** [review]
Thanks!

While adding the type to `test_filter_pushdown.py` is a good first check, the test is quite limited (and should probably be improved)

To test this slightly more extensively can we add it to `test_all_types.py` ?

**Comment by Tishj** [comment]
Also slightly unrelated to this PR, but more generically towards `test_filter_pushdown.py` I made a start on improving the test by combining it with `test_all_types()`

<https://github.com/Tishj/duckdb/tree/python_arrow_filter_pushdown_fix>

What we currently do is hardcoding a table:
```py
    connection.execute(
        f"""
        INSERT INTO {tbl_name} VALUES
            (1,1,1),
            (10,10,10),
            (100,10,100),
            (NULL,NULL,NULL)
    """
    )
```

Which is not testing the ranges of the types properly
By using `test_all_types` to get the values we can use the min and max instead of `1` and `100`

**Comment by Tishj** [comment]
```
D select ('-170141183460469231731687303715884105728'::HUGEINT)::DECIMAL(38,0);
Conversion Error: Could not cast value -170141183460469231731687303715884105728 to DECIMAL(38,0)
LINE 1: ...0469231731687303715884105728'::HUGEINT)::DECIMAL(38,0);
```

This looks like it should be fixed if we're going to export (U)HUGEINT to decimal(38,0) when converting to Arrow

**Comment by carlopi** [comment]
Also, I tried to cherry-pick this on duckdb-wasm side, and this happens:
```
duckdb> SELECT hugeint FROM test_all_types();
┌──────────────────────────────────────────┐
│ hugeint                                  │
╞══════════════════════════════════════════╡
│ -170141183460469231731687303715884105728 │
│  170141183460469231731687303715884105727 │
│                                          │
└──────────────────────────────────────────┘

duckdb> SELECT uhugeint FROM test_all_types();
┌──────────┐
│ uhugeint │
╞══════════╡
│        0 │
│       -1 │
│          │
└──────────┘
```
where hugeint seems to behave correctly, but UHUGEINT for the range 1<<127 / 1<<128 -1 seems to be just represented as negative numbers.

It's not 1-1 that problem is with this PR, but though it might be worth sharing.

File: src/common/arrow/arrow_converter.cpp [hunk 1]
```diff
@@ -124,6 +124,9 @@ void SetArrowFormat(DuckDBArrowSchemaHolder &root_holder, ArrowSchema &child, co
 	case LogicalTypeId::HUGEINT:
 		child.format = "d:38,0";
 		break;
+	case LogicalTypeId::UHUGEINT:
+		child.format = "d:38,0";
```
Comment (Tishj) [inline src/common/arrow/arrow_converter.cpp:128]: One problem with this, for both UHUGEINT and HUGEINT is that their value ranges are bigger than are supported by DECIMAL(38,0)

**Comment by carlopi** [comment]
Fixed by https://github.com/duckdb/duckdb/pull/13576

**Pull request closed without merging**